src/tools/cephfs/cephfs-shell

```diff
@@ -1585,17 +1591,67 @@ if __name__ == '__main__':
             args.commands = ['run_script ' + args.batch, ',quit']
     if args.test:
         args.commands.extend(['-t,'] + [arg + ',' for arg in args.test])
+    if not args.batch and len(args.commands) > 0:
+        args.exe_and_quit = True
+
+    manage_sys_argv(args)
 
+    return args
+
+
+def manage_sys_argv(args):
+    exe = sys.argv[0]
     sys.argv.clear()
     sys.argv.append(exe)
     sys.argv.extend([i.strip() for i in ' '.join(args.commands).split(',')])
 
     setup_cephfs()
+
+
+def execute(args):
+    """
+    Launch a shell session if no arguments were passed, else just execute
+    the given argument as a shell command and exit the shell session
+    immediately at (last) command's termination with the (last) command's
+    return value.
+    """
+    if not args.exe_and_quit:
+        return shell.cmdloop()
+
+    args_to_onecmd = ''
+    if len(args.commands) <= 1:
+        args.commands = args.commands[0].split(' ')
+    # multiple commands might be passed separated by commas, feed onecmd()
+    # one command at a time.
+    for cmdarg in args.commands:
+        if ',' in cmdarg:
+            args_to_onecmd += ' ' + cmdarg[0:-1]
+
+            retval = shell.onecmd(args_to_onecmd)
+            if retval is not None and not retval:
+                return retval
+            if shell.exit_code != 0:
+                return shell.exit_code
+
+            args_to_onecmd = ''
+            continue
+        else:
+            args_to_onecmd += ' ' + cmdarg
+    return shell.onecmd(args_to_onecmd)
+
+
+if __name__ == '__main__':
+    args = manage_args()
+
     shell = CephFSShell()
+    shell.exit_code = 0
```

Comment:
We are setting it later. Why is this one required?

Reply:
It's mistake. Fixed.